wolfcrypt/src/aes.c

```diff
@@ -4628,14 +4628,14 @@ static WC_INLINE void FlattenSzInBits(byte* buf, word32 sz)
     sz <<= 3;
 
     /* copy over the words of the sz into the destination buffer */
-    buf[0] = (szHi >> 24) & 0xff;
-    buf[1] = (szHi >> 16) & 0xff;
-    buf[2] = (szHi >>  8) & 0xff;
-    buf[3] = szHi & 0xff;
-    buf[4] = (sz >> 24) & 0xff;
-    buf[5] = (sz >> 16) & 0xff;
-    buf[6] = (sz >>  8) & 0xff;
-    buf[7] = sz & 0xff;
+    buf[0] = (byte)(szHi >> 24);
```

Comment:
Is it really required to remove the ` & 0xFF`? I prefer that for clarity.

Reply:
the explicit `(byte)` cast does that by definition, and dumb compilers or compilers with optimizers off might, I imagine, frivolously mask twice for the conversion.

in general, doing `& 0xff` then doing explicit `(byte)` is confusing since they do the same thing by definition.